### ISSUE TITLE
Muon reconstruction failing

### ISSUE THREAD

**Issue by rete** [iLCSoft/ILDConfig#88]
We see a non-negligible fraction of single muon events failing. This issue was first communicated by email: 

```
Dear all,

now to the next problem, which concerns muons around costh~ -0.6 (and also +0.6, but I'm not yet convinced that these are identical...)

in single particle 20 gev mu- events, we sometimes get exactly 1 PandoraPFO, which is neutral and contains only MUON hits.
The MarlinTrack is there, as are ECAL/HCAL hits, but they don't make it into a PandoraPFO.
I think there may be a related effect at 90deg: at least the symptoms look similar.

I stripped off some events which created problems, and uploaded them here:   http://research.kek.jp/people/jeans/badmuons/
You can find the sim output, reco output, and reco logfile (with MarlinPandora in DEBUG mode, which shows some complaints/errors).

The behaviour in physics (ie multi-particle) events is different. If I understand correctly, Junping has noticed that his muons sometime/often result in charged PFOs without assigned muon hits, in the similar/same costh region.
```

The problematic error stack is the following:

```
[ VERBOSE "MyDDMarlinPandora"] ClusterFitHelper: linear fit to cluster failed. 
[ VERBOSE "MyDDMarlinPandora"] ClusterFitHelper::FitStart(pCluster, m_nClusterLayersToFit, clusterFitResult) return STATUS_CODE_INVALID_PARAMETER
[ VERBOSE "MyDDMarlinPandora"]     in function: AssociateMuonTracks
[ VERBOSE "MyDDMarlinPandora"]     in file:     /cvmfs/ilc.desy.de/sw/x86_64_gcc49_sl6/v02-00-01/PandoraPFANew/v03-09-00/LCContent-v03-01-03/src/LCParticleId/MuonReconstructionAlgorithm.cc line#: 96
[ VERBOSE "MyDDMarlinPandora"] this->AssociateMuonTracks(pMuonClusterList) return STATUS_CODE_INVALID_PARAMETER
[ VERBOSE "MyDDMarlinPandora"]     in function: Run
[ VERBOSE "MyDDMarlinPandora"]     in file:     /cvmfs/ilc.desy.de/sw/x86_64_gcc49_sl6/v02-00-01/PandoraPFANew/v03-09-00/LCContent-v03-01-03/src/LCParticleId/MuonReconstructionAlgorithm.cc line#: 55
[ VERBOSE "MyDDMarlinPandora"] iter->second->Run() throw STATUS_CODE_INVALID_PARAMETER
[ VERBOSE "MyDDMarlinPandora"]     in function: RunAlgorithm
[ VERBOSE "MyDDMarlinPandora"]     in file:     /cvmfs/ilc.desy.de/sw/x86_64_gcc49_sl6/v02-00-01/PandoraPFANew/v03-09-00/PandoraSDK-v03-02-01/src/Api/PandoraContentApiImpl.cc line#: 231
[ VERBOSE "MyDDMarlinPandora"] Failure in algorithm Alg0003, MuonReconstruction, STATUS_CODE_INVALID_PARAMETER
[ VERBOSE "MyDDMarlinPandora"] PhotonReconstructionAlgorithm::CreateRegionsOfInterests - no photon candidates available, no regions of interests created.
[ VERBOSE "MyDDMarlinPandora"] HighEnergyPhotonRecoveryAlgorithm: Failed to obtain cluster list PhotonClusters
[ VERBOSE "MyDDMarlinPandora"] SoftClusterMergingAlgorithm: Failed to obtain cluster list PhotonClusters
[ VERBOSE "MyDDMarlinPandora"] IsolatedHitMergingAlgorithm: Failed to obtain cluster list PhotonClusters
[ VERBOSE "MyDDMarlinPandora"] PandoraContentApi::ReplaceCurrentList<CaloHit>(*this, m_inputCaloHitListName) return STATUS_CODE_NOT_FOUND
[ VERBOSE "MyDDMarlinPandora"]     in function: Run
[ VERBOSE "MyDDMarlinPandora"]     in file:     /cvmfs/ilc.desy.de/sw/x86_64_gcc49_sl6/v02-00-01/PandoraPFANew/v03-09-00/LCContent-v03-01-03/src/LCClustering/ClusteringParentAlgorithm.cc line#: 34
[ VERBOSE "MyDDMarlinPandora"] iter->second->Run() throw STATUS_CODE_NOT_FOUND
[ VERBOSE "MyDDMarlinPandora"]     in function: RunAlgorithm
[ VERBOSE "MyDDMarlinPandora"]     in file:     /cvmfs/ilc.desy.de/sw/x86_64_gcc49_sl6/v02-00-01/PandoraPFANew/v03-09-00/PandoraSDK-v03-02-01/src/Api/PandoraContentApiImpl.cc line#: 231
[ VERBOSE "MyDDMarlinPandora"] Failure in algorithm Alg0075, ClusteringParent, STATUS_CODE_NOT_FOUND
```

The problem comes from the cluster fit failing thus making the muon clustering algorithm failing too.

The different resources for debugging, as pointed out from the mail are available here:
- http://research.kek.jp/people/jeans/badmuons/

**Comment by rete** [comment]
Related issue: https://github.com/iLCSoft/lcgeo/issues/227

**Comment by rete** [comment]
Related fix: #89
